pkg/helpers/image/dockerlayer/add/add.go

```diff
@@ -52,7 +52,7 @@ func DigestCopy(dst io.ReaderFrom, src io.Reader) (layerDigest, blobDigest diges
 		defer close(ch)
 		gr, err := gzip.NewReader(pr)
 		if err != nil {
-			ch <- fmt.Errorf("unable to create gzip reader layer upload: %v", err)
+			pr.CloseWithError(fmt.Errorf("unable to create gzip reader layer upload: %v", err))
```

Comment:
My only concern is that gzip.NewReader may have already read all data from `pr` at this point (for example, when there were less than 10 bytes to read). In this case this error won't be detected as nothing will write into the writer.

Reply:
Yes, you are absolutely right. I think it makes sense to still send the error down through that error channel. I have updated this to just close the pipe reader end with an error.